### PR TITLE
AK+LibJS+Meta: Implement and use Unicode case folding

### PR DESCRIPTION
**Pull request by trflynn89**
This adds a Lagom utility to `Meta/Unicode` for parsing [UnicodeData.txt](https://www.unicode.org/Public/13.0.0/ucd/UnicodeData.txt) and generating C++ to hold parsed data.

UnicodeData.txt is a pretty large file. So rather than committing that file to git and generating the C++ into `Build/i686/AK`, this utility is meant to be run manually to generate the C++ directly into `AK` on an as-needed basis. But if we want to commit UnicodeData.txt to git, then running this utility could be part of the normal build process.

A regex/JS feature I'm hoping to get from this process is Unicode property escapes - there's about 550 test262 tests up for grabs there :)

### CODE REVIEW

**Comment by alimpfard** [comment]
Let's...not include the generated file perhaps?
you could make cmake download the unicode data file and generate the file?

**Comment by linusg** [comment]
> But if we want to commit UnicodeData.txt to git, then running this utility could be part of the normal build process.

I'm strongly against adding any large/external files to the repo, that can (and should) be downloaded (or copied into the build dir manually if working offline) as part of the build process. Related: #5627

**Review by linusg — CHANGES_REQUESTED** [review]
Let's generate `AK/UnicodeData.cpp` at build time and remove the committed file.

File: Meta/Unicode/main.cpp [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    bool success = file_or_error.value()->write(generator.as_string_view());
+    VERIFY(success);
+}
+
+void generate_unicode_data_implementation(UnicodeData unicode_data, String implementation_path)
+{
+    StringBuilder builder;
+    SourceGenerator generator { builder };
+
+    generator.set("size", String::formatted("{}", unicode_data.code_point_data.size()));
```
Comment (IdanHo) [inline Meta/Unicode/main.cpp:189]: `String::number()` would be nicer here

**Review by IdanHo — COMMENTED** [review]
Great work on this, this will make adding more unicode support to LibJS and LibGfx much easier :) As linus and ali already suggested, UnicodeData.txt should be downloaded and cached on compile time similarly to PCI/USB IDs, and the source files should be generated at that point as well similarly to IPC or LibWeb bindings.

**Comment by trflynn89** [comment]
Closing in favor of #9004 (was easier to just start over on a fresh branch) :)

**Pull request closed without merging**